### PR TITLE
Enhancement/98/refactoring

### PR DESCRIPTION
**Pull request by xcaverxiii**
Related issue #98

Refactor some angular components to better code .
All features should work as before.

### CODE REVIEW

**Comment by xcaverxiii** [comment]
fixed iconclass conflicts

**Comment by xcaverxiii** [comment]
fixed merge conflicts ... again...

File: app/src/app/core/services/data.service.ts [hunk 1]
```diff
@@ -193,11 +194,8 @@ export class DataService {
    * @returns an Array containing the iconclassData to the respective Iconclass
    */
   public async getIconclassData(iconclasses:Array<Iconclass>): Promise<any> {
-    const results : Array<any> = [];
-    await Promise.all(iconclasses.map(async (key:Iconclass) => 
-      results.push(await this.http.get(`http://iconclass.org/${key}.json`).toPromise())
-    ));
-    return results;
+    return await iconclasses.map(async (key:Iconclass) =>
```
Comment (mxxbe) [inline app/src/app/core/services/data.service.ts]: have you tested this with an array containing more than one iconclass?

Comment (xcaverxiii) [reply app/src/app/core/services/data.service.ts]: No i haven´t. But you could right, that could be a problem. I readded Promise all it should be wait for all async functions in this map.

**Pull request merged**